src/component/component.ts

```diff
@@ -579,10 +577,14 @@ export class Component<T extends Env, Props extends {}> {
       __owl__.observer.allowMutations = false;
     }
     try {
-      fiber.vnode = __owl__.renderFn!(this, {
+      let vnode = __owl__.renderFn!(this, {
         handlers: __owl__.boundHandlers,
         fiber: fiber
       });
+      if (!vnode) {
+        throw new Error(`Rendering '${this.constructor.name}' did not return a vnode`);
```

Comment:
Isn't the vnode stuff an implementation detail? Actually, the problem here is that the component has an empty template